### PR TITLE
Feature: tlsPassClientCertificates filter

### PR DESCRIPTION
**Pull request by szuecs**
https://github.com/zalando/skipper/issues/1780#issuecomment-1926971779 showed interest in a filter that passes TLS client cert from the request to the proxy as header to the backend similar to a traefik feature. I trimmed it down to pem only support, but I think it's the one that is interesting without too much configuration.

Cc @containerpope

### CODE REVIEW

File: filters/tls/pass_client.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+const (
+	certSeparator           = ","
+	xForwardedTLSClientCert = "X-Forwarded-Tls-Client-Cert"
+)
+
+// sanitize the raw certificates, remove the useless data and make it http request compliant.
+func sanitize(cert []byte) string {
+	return strings.NewReplacer(
```
Comment (AlexanderYastrebov) [inline filters/tls/pass_client.go:37]: strings.NewReplacer documented to be safe for concurrent use by multiple goroutines, maybe it makes sense to have one global instance

Comment (szuecs) [reply filters/tls/pass_client.go:37]: makes sense, I just copied the code more or less as it was, but does not hurt to make it better :)

Comment (szuecs) [reply filters/tls/pass_client.go]: done

File: filters/tls/pass_client.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	if len(args) != 0 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	return &tlsFilter{}, nil
+}
+
+const (
+	certSeparator           = ","
+	xForwardedTLSClientCert = "X-Forwarded-Tls-Client-Cert"
```
[the remark this answers is not in the document]
Comment (szuecs) [reply filters/tls/pass_client.go]: done

File: docs/reference/filters.md [hunk 3]
```diff
@@ -450,6 +450,21 @@ The second filter will set `Authorization` header to the
 `access_token` query param with a prefix value `Bearer ` and will
 not override the value if the header exists already.
 
+## TLS
+
+Filters that provide access to TLS data of a request.
+
+### tlsPassClient
+
+This filter copies TLS client certificate encoded as pem into the
```
Comment (AlexanderYastrebov) [inline docs/reference/filters.md:459]: I think it could be multiple certificates, no?

File: docs/reference/filters.md [hunk 4]
```diff
@@ -450,6 +450,21 @@ The second filter will set `Authorization` header to the
 `access_token` query param with a prefix value `Bearer ` and will
 not override the value if the header exists already.
 
+## TLS
+
+Filters that provide access to TLS data of a request.
+
+### tlsPassClient
```
Comment (AlexanderYastrebov) [inline docs/reference/filters.md:457]: `tlsPassClientCertificate`

File: docs/reference/filters.md (hunk 3 above)
Comment (szuecs) [reply docs/reference/filters.md:459]: yes true

Comment (szuecs) [reply docs/reference/filters.md]: done

File: docs/reference/filters.md (hunk 4 above)
Comment (szuecs) [reply docs/reference/filters.md]: changed

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**